api/policies/hasValidUsername.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * hasValidUsername
+ *
+ * @module      :: Policy
+ * @description :: Only allows requests that contain a 'username' parameter, which is a string and does not contain spaces
+ * @docs        :: http://sailsjs.org/#!/documentation/concepts/Policies
+ *
+ */
+module.exports = function (req, res, next) {
+  if (req.body.hasOwnProperty('username')) {
+    const username = req.body.username;
+    if (typeof (username) === 'string') {
+      if (username.match(/^[0-9a-zA-Z.@]+$/)){ // . and @ are included to allow for emails
+        return next();
+      } else {
+        if (username.length === 0) {
+          return res.badRequest({message: 'Please provide a valid username'});
+        }
+        return res.badRequest({message: "Your username must contain only letters or numbers"});
```

Comment:
If we are going to accept @'s etc we should adjust this error message accordingly. I'm not actually sure what's reasonable to require. No empty strings, idk what else tbh. Thoughts?

Reply:
I think we should accept @s just so people can continue to use what they had before; as for new names, I think it's ok if we don't say exactly what can be in it, just letters and numbers?